### PR TITLE
Make Unix socket writable for other user (says, Nginx)

### PR DESCRIPTION
**Pull request by hongquan**
## What do these changes do?

Fix the permission of Unix socket domain created by `web.run_app(path=...)`

## Are there changes in behavior for the user?

No

## Related issue number

No

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by asvetlov** [comment]
I suspect that `o666` mode is highly insecure

**Comment by hongquan** [comment]
> As Andrew wrote earlier, `666` is highly insecure and we cannot accept this change as is.
> If you change it to be configurable with a user setting, however, this would be acceptable.

@asvetlov , @webknjaz 
Why is it insecure, when it is a socket?
If you replace Unix socket with TCP/IP one, every one can "send" to the port, equivalent with "write" to Unix socket.
Why in TCP/IP socket form, you don't concern about "everyone can write" fact, then suddenly in Unix socket, you worry about it?

**Comment by webknjaz** [comment]
@hongquan UNIX file socket is a node in the file system. Write privileges mean that it can be unlinked/replaced/hijacked if not protected by proper privileges or more advanced tools like SELinux-based ACL attributes.

File: aiohttp/web_runner.py [hunk 1]
```diff
@@ -135,6 +136,9 @@ def name(self) -> str:
         self._server = await loop.create_unix_server(
             server, self._path,
             ssl=self._ssl_context, backlog=self._backlog)
+        # create_unix_server sets file permission as rwxr-xr-x, which is not
```
Comment (webknjaz) [inline aiohttp/web_runner.py:139]: @hongquan this assumption is incorrect. `create_unix_server()` doesn't set any privileges. It's the OS kernel which does so following the currently set `umask` value. You can modify the umask in the shell session where you run aiohttp and it file creation syscalls will respect that.

File: aiohttp/web_runner.py [hunk 2]
```diff
@@ -135,6 +136,9 @@ def name(self) -> str:
...
+        # helpful in practice, so we fix it.
+        os.chmod(self._path, 0o664)
```
Comment (webknjaz) [inline aiohttp/web_runner.py:141]: Based on the above, I still want it to be configurable. Also, the default should give write privileges only to the owner.

File: aiohttp/web_runner.py [hunk 3]
```diff
@@ -135,6 +136,9 @@ def name(self) -> str:
         self._server = await loop.create_unix_server(
             server, self._path,
             ssl=self._ssl_context, backlog=self._backlog)
+        # create_unix_server sets file permission as rwxr-xr-x, which is not
+        # helpful in practice, so we fix it.
```
Comment (webknjaz) [inline aiohttp/web_runner.py:140]: It's not broken. Nothing to fix here.

File: aiohttp/web_runner.py (hunk 1 above)
Comment (hongquan) [reply aiohttp/web_runner.py:139]: Thanks.

**Comment by hongquan** [comment]
I'm closing this PR because "configurable file mode" is preferred.

**Comment by guusbertens** [comment]
@hongquan @webknjaz You're wrong about the permissions: whether you can remove a unix socket (or any file) depends on the permission bits that are set on the containing directory. Simply put: if you own the directory, you can remove files/sockets no matter who owns them. This is not the case if the sticky bit is set; see e.g. [here](https://linux.die.net/man/1/chmod) or [here](https://www.freebsd.org/cgi/man.cgi?query=sticky&sektion=7&apropos=0&manpath=FreeBSD+12.1-RELEASE+and+Ports).
In other words: putting a socket with 0o666 in /tmp (on which most distros set the sticky bit) means that other users cannot remove it.
Is there still interest in having a "configurable file mode"? If I get positive response, I can see what I can do.

**Comment by hongquan** [comment]
@guusbertens If you confirm that a Unix socket file cannot be deleted by non-owner user, then it is better to make Unix writeable for other user (convention over configuration). The sticky bit of parent directory can be set by other mean (deploy script).

My favourite deployment way is that, I let systemd create a dedicated folder for my sockets, via `WorkingDirectory` option. With this, we can control the folder sticky bit, then all my sockets inside can be safe from deleting by other users, but let Nginx "write" to it.
